precompiles/erc20/integration_test.go

```diff
@@ -179,7 +181,405 @@ var _ = Describe("ERC20 Extension -", func() {
 		_ = passCheck
 	})
 
-	Context("basic functionality -", func() {})
+	Context("basic functionality -", func() {
+		When("transferring tokens", func() {
+			DescribeTable("it should transfer tokens to a non-existing address", func(callType CallType, expGasUsed int64) {
+				sender := is.keyring.GetKey(0)
+				receiver := utiltx.GenerateAddress()
+				fundCoins := sdk.Coins{sdk.NewInt64Coin(is.tokenDenom, 300)}
+				transferCoins := sdk.Coins{sdk.NewInt64Coin(is.tokenDenom, 100)}
+
+				// Fund account with some tokens
+				is.fundWithTokens(callType, contractsData, sender.Addr, fundCoins)
+
+				// Transfer tokens
+				txArgs, transferArgs := is.getTxAndCallArgs(callType, contractsData, erc20.TransferMethod, receiver, transferCoins[0].Amount.BigInt())
+
+				transferCheck := passCheck.WithExpEvents(erc20.EventTypeTransfer)
+
+				res, _, err := is.factory.CallContractAndCheckLogs(sender.Priv, txArgs, transferArgs, transferCheck)
+				Expect(err).ToNot(HaveOccurred(), "unexpected result calling contract")
+
+				is.ExpectBalancesForContract(
+					callType, contractsData,
+					[]ExpectedBalance{
+						{address: sender.AccAddr, expCoins: fundCoins.Sub(transferCoins...)},
+						{address: receiver.Bytes(), expCoins: transferCoins},
+					},
+				)
+
+				Expect(res.GasUsed).To(Equal(expGasUsed), "expected different gas used")
+			},
+				// FIXME: The gas used on the precompile is much higher than on the EVM
+				Entry(" - direct call", directCall, int64(3_021_572)),
```

Comment:
We need to fix this, 3M gas is extremely expensive

Reply:
will adjust on a different PR 👍 I want to gather average gas consumption for all of the methods throughout the day and then create one PR based on the findings